src/OmniSharp.Stdio/Program.cs

```diff
@@ -0,0 +1,264 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Composition.Hosting;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Endpoint;
+using OmniSharp.Eventing;
+using OmniSharp.Mef;
+using OmniSharp.Models.UpdateBuffer;
+using OmniSharp.Plugins;
+using OmniSharp.Services;
+using OmniSharp.Stdio.Logging;
+using OmniSharp.Stdio.Protocol;
+using OmniSharp.Stdio.Services;
+using OmniSharp.Utilities;
+
+namespace OmniSharp.Stdio
+{
+    public class Program : IDisposable
+    {
+        public static int Main(string[] args) => OmniSharp.Start(() =>
+        {
+            var application = new OmniSharpStdioCommandLineApplication();
+            application.OnExecute(() =>
+            {
+                var environment = application.CreateEnvironment();
+                var writer = new SharedConsoleWriter();
+                var plugins = application.CreatePluginAssemblies();
+                var configuration = new OmniSharpConfigurationBuilder(environment).Build();
+                var serviceProvider = OmniSharpMefBuilder.CreateDefaultServiceProvider(configuration);
+                var mefBuilder = new OmniSharpMefBuilder(serviceProvider, environment, writer, new StdioEventEmitter(writer));
+                var compositionHost = mefBuilder.Build();
+                var loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
+                var cancellation = new CancellationTokenSource();
+
+                using (var program = new Program(Console.In, writer, environment, configuration, serviceProvider, compositionHost, loggerFactory, cancellation))
```

Comment:
this feels a bit weird IMHO, a `Main` in a `Program`, that then instantiates `Program` and calls another "Main" (`Start`) on it. I think for sanity reasons it should be split into dedicated classes, and let `Program` just act as an entry point